sync15-adapter/src/state.rs

```diff
@@ -46,3 +73,568 @@ impl GlobalState {
+
+    struct InMemoryClient {
+        info_configuration: error::Result<InfoConfiguration>,
+        info_collections: error::Result<HashMap<String, ServerTimestamp>>,
+        meta_global: error::Result<BsoRecord<MetaGlobalRecord>>,
+        crypto_keys: error::Result<BsoRecord<EncryptedPayload>>,
+    }
+
+    impl SetupStorageClient for InMemoryClient {
+        fn fetch_info_configuration(&self) -> error::Result<InfoConfiguration> {
+            match &self.info_configuration {
+                Ok(config) => Ok(config.clone()),
+                Err(_) => Err(ErrorKind::StorageHttpError {
+                    code: reqwest::StatusCode::InternalServerError,
+                    route: "info/configuration".to_string(),
+                }.into()),
+            }
+        }
+
+        fn fetch_info_collections(&self) -> error::Result<HashMap<String, ServerTimestamp>> {
+            match &self.info_collections {
+                Ok(collections) => Ok(collections.clone()),
+                Err(_) => Err(ErrorKind::StorageHttpError {
+                    code: reqwest::StatusCode::InternalServerError,
+                    route: "info/collections".to_string(),
+                }.into()),
+            }
+        }
+
+        fn fetch_meta_global(&self) -> error::Result<BsoRecord<MetaGlobalRecord>> {
+            match &self.meta_global {
+                Ok(global) => Ok(global.clone()),
+                // TODO(lina): Special handling for 404s, we want to ensure we
+                // handle missing keys and other server errors correctly.
+                Err(_) => Err(ErrorKind::StorageHttpError {
+                    code: reqwest::StatusCode::InternalServerError,
+                    route: "meta/global".to_string(),
+                }.into()),
+            }
+        }
+
+        fn put_meta_global(&self, global: &BsoRecord<MetaGlobalRecord>) -> error::Result<()> {
+            Err(ErrorKind::StorageHttpError {
+                code: reqwest::StatusCode::InternalServerError,
+                route: "meta/global".to_string(),
+            }.into())
+        }
+
+        fn fetch_crypto_keys(&self) -> error::Result<BsoRecord<EncryptedPayload>> {
+            match &self.crypto_keys {
+                Ok(keys) => Ok(keys.clone()),
+                // TODO(lina): Same as above, for 404s.
+                Err(_) => Err(ErrorKind::StorageHttpError {
+                    code: reqwest::StatusCode::InternalServerError,
+                    route: "crypto/keys".to_string(),
+                }.into()),
+            }
+        }
+
+        fn put_crypto_keys(&self, keys: &EncryptedBso) -> error::Result<()> {
+            Err(ErrorKind::StorageHttpError {
+                code: reqwest::StatusCode::InternalServerError,
+                route: "crypto/keys".to_string(),
+            }.into())
+        }
+
+        fn wipe_all(&self) -> error::Result<()> {
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn test_state_machine() {
```

Comment:
This is really only testing the happy path, so let's call it something more specific.